File: packages/mapo-api/files/file_model.go
package files

import "fmt"

type File struct {
	ID             int    `json:"id"`
	UserID         string `json:"userId"`
	Name           string `json:"name"`
	ContentBase64  string `json:"contentBase64"`
	FolderID       int    `json:"folderId"` // NOTE: We use -1 to represent the root folder, but there's no real entry with that id
	CreatedAt      string `json:"createdAt"`
	LastModifiedAt string `json:"lastModifiedAt"`
}

func (f File) String() string {
	return fmt.Sprintf("File{ id: %d, user_id: %s, name: %s, content: %s, createdAt: %s, lastModifiedAt: %s }",
		f.ID, f.UserID, f.Name, f.ContentBase64, f.CreatedAt, f.LastModifiedAt)
}

File: packages/mapo-webapp/src/app/models/file.model.ts
import { z } from 'zod';
import { EdgeSchema } from './edge.model';
import { TextNodeSchema } from './textnode.model';

/**
 * A Mapo file, storing a single drawing for a user
 */
export const FileSchema = z.object({
  id: z.number(),
  userId: z.string(),
  name: z.string(),
  contentBase64: z.string(),
  folderId: z.number().describe('ID of the parent folder, where 0 is the root folder.')
});

export type File = z.infer<typeof FileSchema>;

/**
 * The content of a file that needs to be loaded when we render a file on the canvas
 * Also what get saved in local storage
 */
export const FileContentSchema = z.object({
  id: z.number().nullable(),
  name: z.string(),
  edges: z.array(EdgeSchema),
  textNodes: z.array(TextNodeSchema),
  title: z.string().optional().describe('Legacy version of \'name\', kept for backwards compatibility')
});

export type FileContent = z.infer<typeof FileContentSchema>;
